### ISSUE TITLE
yarn pack throws toString failed error

### ISSUE THREAD

**Issue by sarus** [yarnpkg/yarn#928]
**Do you want to request a _feature_ or report a _bug_?**

Bug

**What is the current behavior?**

When running `yarn pack` it runs for several minutes and then I get the following error:

```
buffer.js:440
    throw new Error('toString failed');
    ^

Error: toString failed
    at Buffer.toString (buffer.js:440:11)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:378:21)
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Run `yarn pack` on our project.  I understand the error doesn't give much to go on.  If there is some way to get a more detailed error message we'd be happy to rerun the command and provide more details.   

**What is the expected behavior?**

`yarn park` would successfully create a tgz file.

**Please mention your node.js, yarn and operating system version.**

```
node v4.5.0
yarn v0.15.0
CentOS Linux release 7.1.1503 (Core)
```

 Thanks!

**Comment by FLGMwt** [comment]
Could you dump the full stack (if there's more after `readFileAfterClose`) and maybe the package.json?

**Comment by sarus** [comment]
@FLGMwt Is there a special flag to get a full stack trace to print?  I saw if I was running node directly I could maybe try something like `--stack_trace_limit`.  What I pasted above is everything that was output to console.  

In regards to the package.json is there a particular property you are looking for?  The one things that stands out to me as perhaps being "non-typical" is that we use several private repos with tags in the dependency list of the form:

```
"privateDep": "ssh://git@bitbucket.org/sarus/privateDep.git#v0.1.1",
```

Note that these were originally of the form:

```
"privateDep": "git+ssh://git@bitbucket.org:sarus/privateDep.git#v0.1.1",
```

However, I had to change them over because of https://github.com/yarnpkg/yarn/issues/513

`yarn install` did complete successfully once I changed over the format of our private repos.
